tests/Unit/Controllers/Store/BundleControllerTest.php

```diff
@@ -133,6 +133,77 @@ public function testICannotSubmitInvalidValuesToAppendVouchers()
         }
     }
 
+    /** @test */
+    public function testIMustDisburseWithAllRelevantFields()
+    {
+        $dataSets = [
+            [
+                "data" => [
+                    "collected_at" => "1", "collected_on" => "2018-07-21"
+                ],
+                "outcome" => [ "collected_by" => "The collected by field is required when collected at / collected on is present."],
+            ],
+            [
+                "data" => [
+                    "collected_by" => "1", "collected_on" => "2018-07-21"
+                ],
+                "outcome" => [ "collected_at" => "The collected at field is required when collected on / collected by is present."],
+            ],
+            [
+                "data" => [
+                    "collected_at" => "1", "collected_by" => "1"
+                ],
+                "outcome" => [ "collected_on" => "The collected on field is required when collected at / collected by is present."],
+            ],
+            [
+                "data" => [
+                    "collected_at" => "1", "collected_on" => "invalid", "collected_by" => "1"
+                ],
+                "outcome" => [ "collected_on" => "The collected on does not match the format Y-m-d."],
+            ],
+            [
+                "data" => [
+                    "collected_at" => "9999", "collected_on" => "2018-07-21", "collected_by" => "1"
+                ],
+                "outcome" => [ "collected_at" => "The selected collected at is invalid."],
+            ],
+            [
+                "data" => [
+                    "collected_at" => "1", "collected_on" => "2018-07-21", "collected_by" => "9999"
+                ],
+                "outcome" => [ "collected_by" => "The selected collected by is invalid."],
+            ],
+        ];
+
+        $route = route('store.registration.voucher-manager', [ 'registration' => $this->registration->id ]);
+        $put_route = route('store.registration.vouchers.put', ['registration' => $this->registration->id]);
+
+        foreach ($dataSets as $set) {
+            $response = $this->actingAs($this->centreUser, 'store')
+                ->visit($route)
+                ->put(
+                    $put_route,
+                    $set["data"]
+                )
+            ;
+            // work out which field we're testing.
+            $field = array_keys($set['outcome'])[0];
+
+            // Dig out errors from Session
+            $response->seeInSession('errors');
+            $errors = Session::get("errors")->get($field);
+
+            // Check our specific message is present
+            $this->assertContains($set['outcome'][$field], $errors);
+
+            // we follow that to the correct page;
+            $this->followRedirects()
+                ->seePageIs($route)
+                ->assertResponseStatus(200)
```

Comment:
Should it be 200 if we are looking at errors?

Reply:
It's a redirect with form errors, so I'm hoping to land back on the page. At this point I can't check that the errors are in the html, because we haven't decided how/if to show them.